modules/instruments/php/visitsummary.class.inc

```diff
@@ -0,0 +1,135 @@
+<?php declare(strict_types=1);
+
+namespace LORIS\instruments;
+
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \LORIS\StudyEntities\Candidate\CandID;
+
+/**
+ * The VisitSummary endpoint returns a summary of instrument
+ * data entry for a particular candidate's visit. This is primarily
+ * to be used by the instrument list widget.
+ *
+ * @license http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ */
+class VisitSummary extends \NDB_Page
+{
+    /**
+     * Temp for testing.
+     *
+     * @param \User $user The user whose access is to be verified
+     *
+     * @return bool
+     */
+    function _hasAccess(\User $user) : bool
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        $user = $request->getAttribute("user");
+        if ($user === null && !($user instanceof \User)) {
+            return new \LORIS\Http\Response\JSON\InternalServerError(
+                "No valid user"
+            );
+        }
```

Comment:
I'm a little confused about this part. Doesn't the Middleware do something to detect whether the User is logged in or whether it's anonymous/Unauthorized? Why is this being checked here in particular when it isn't in other modules?

Reply:
We only have middleware to add appropriate page decorations depending on whether someone is logged in. The parent does access checking, but the parent isn't called here because we only want to return JSON responses.

At any rate, this check is here because without it our static analysis tools (I think it was phan, but it might have been one of the other ones) can't deduce that `$user` is an instance of `\User`. It should always be added to the request by the router, that's why it's an internal server error if it's not.

Reply:
Fair enough. It might be overkill just to fix phan but I believe if we were to extend `ServerRequestInterface` (e.g. `LORISRequest`) and add a typed `getUser()` function then we could address that issue.